src/test/java/com/microsoft/sqlserver/jdbc/unit/serial/DTOSerialTest.java

```diff
@@ -0,0 +1,199 @@
+
+        // create a DTO
+        try (Connection conn = DriverManager.getConnection(connectionString);
+                SQLServerPreparedStatement ps = (SQLServerPreparedStatement) conn
+                        .prepareStatement("SELECT CAST(? AS datetimeoffset(7)) AS" + "   'datetimeoffset IS08601' ")) {
+            ps.setDateTimeOffset(1, dtN);
+            ResultSet rs = ps.executeQuery();
+            rs.next();
+            verifyDTOEqual(dtN, ((SQLServerResultSet) rs).getDateTimeOffset(1));
+        }
+    }
+
+    // Negative test cases.
+    private static void verifyMessedSerialization() throws Exception {
+        // serialized DTO class with wrong nano values (-1)
+        byte wrongNanos[] = {-84, -19, 0, 5, 115, 114, 0, 47, 109, 105, 99, 114, 111, 115, 111, 102, 116, 46, 115, 113,
+                108, 46, 68, 97, 116, 101, 84, 105, 109, 101, 79, 102, 102, 115, 101, 116, 36, 83, 101, 114, 105, 97,
+                108, 105, 122, 97, 116, 105, 111, 110, 80, 114, 111, 120, 121, 9, 57, 90, 0, -49, -42, -72, 50, 2, 0, 3,
+                73, 0, 13, 109, 105, 110, 117, 116, 101, 115, 79, 102, 102, 115, 101, 116, 73, 0, 5, 110, 97, 110, 111,
+                115, 74, 0, 9, 117, 116, 99, 77, 105, 108, 108, 105, 115, 120, 112, 0, 0, 3, 12, -1, -1, -1, -1, 0, 0,
+                0, 0, 0, 0, 0, 1};
+
+        // serialized DTO class with wrong offset (15*60)
+        byte wrongOffset[] = {-84, -19, 0, 5, 115, 114, 0, 47, 109, 105, 99, 114, 111, 115, 111, 102, 116, 46, 115, 113,
+                108, 46, 68, 97, 116, 101, 84, 105, 109, 101, 79, 102, 102, 115, 101, 116, 36, 83, 101, 114, 105, 97,
+                108, 105, 122, 97, 116, 105, 111, 110, 80, 114, 111, 120, 121, 9, 57, 90, 0, -49, -42, -72, 50, 2, 0, 3,
+                73, 0, 13, 109, 105, 110, 117, 116, 101, 115, 79, 102, 102, 115, 101, 116, 73, 0, 5, 110, 97, 110, 111,
+                115, 74, 0, 9, 117, 116, 99, 77, 105, 108, 108, 105, 115, 120, 112, 0, 0, 3, -124, 0, 0, 1, 44, 0, 0, 0,
+                0, 0, 0, 0, 1};
+
+        // These two serialized forms throw the exception IllegalArgumentException
+        boolean exThrown = false;
+        try {
+            verifyMessedSerializationHelper(wrongNanos);
+        } catch (IllegalArgumentException e) {
+            exThrown = true;
+        }
+
+        if (!exThrown) {
+            fail("serialized form with wrong nano values unexpectedly succeeded");
+        }
+
+        exThrown = false;
+        try {
+            verifyMessedSerializationHelper(wrongOffset);
+        } catch (IllegalArgumentException e) {
+            exThrown = true;
+        }
+
+        if (!exThrown) {
+            fail("serialized form with wrong offset unexpectedly succeeded");
+        }
+    }
+
+    private static void verifyMessedSerializationHelper(byte[] svalue) throws Exception {
+        ObjectInputStream in = new ObjectInputStream(new ByteArrayInputStream(svalue));
+        DateTimeOffset dtn = (DateTimeOffset) in.readObject();

```

Comment:
What is the purpose reading from these Streams? Should validate `dtn`?

Reply:
added comment, this will throw error for negative tests